packages/channel/src/buffers.ts

```diff
@@ -1,95 +1,102 @@
-export interface ChannelBuffer<T> {
-  full: boolean;
-  empty: boolean;
-  add(value: T): void;
-  remove(): T;
+import {
+  CannotReadFromEmptyBufferError,
+  CannotWriteToFullBufferError,
+  InvalidBufferCapacityError,
+} from "./errors";
+
+export abstract class ChannelBuffer<T> {
```

Comment:
I don’t like the idea of making ChannelBuffer an abstract class because it forces all buffers to have a `capacity` property which might not be used, depending on the implementation. I would rather leave it as an interface.